javagen/src/main/java/com/azure/autorest/mapper/ClientParameterMapper.java

```diff
@@ -45,11 +46,14 @@ public ClientMethodParameter map(Parameter parameter) {
         if (parameter.getSchema() != null && parameter.getSchema().getLanguage() != null) {
             parameterDescription = parameter.getSchema().getLanguage().getDefault().getDescription();
         }
+        String name = parameter.getOriginalParameter() != null
+                ? CodeNamer.toCamelCase(parameter.getOriginalParameter().getSchema().getLanguage().getJava().getName()) + CodeNamer.toPascalCase(parameter.getLanguage().getJava().getName())
```

Comment:
Modelerfour upgrade causing the flattened parameter name taking same name as the original parameter name. Had to change it a bit.